Improve 'first wagtail site' tutorial (#9329)
wagtail/wagtail: docs/getting_started/tutorial.md

### Generate your site

Wagtail provides a `start` command similar to `django-admin startproject`.

> Running `wagtail start mysite` in your project will generate:
> - a new `mysite` folder containing the required project settings with a few Wagtail-specific extras,
> - a "home" app with a blank `HomePage` model and basic templates,
> - a sample "search" app
> - and other dependencies for wagtail to run in the the project.

Reviewer: This doesn’t seem that much clearer to me.

```suggestion
Running `wagtail start mysite` in your project will generate a new `mysite` folder with a few Wagtail-specific extras, including
the required project settings,
a "home" app with a blank `HomePage` model and basic templates,
and a sample "search" app.
```

> Appending `mysite` (which is a folder that was created when we ran `python3 -m venv mysite\env`) to `wagtail start mysite` specifies the destination directory for the above items. So the code looks like this:

Reviewer: I quite like the idea of changing this but I don’t think the proposed rewrite is that much clearer. I think I’d prefer we just change the venv folder name.

```suggestion
Because the folder `mysite` was already created by `venv`, run `wagtail start` with an additional argument to specify the destination directory:
```

```sh
wagtail start mysite mysite
```

```{note}

> Generally, in Wagtail, each page type, or content type, is represented by a single app. However, different apps can be aware of each other and access each other's data. All of the apps need to be registered within the `INSTALLED_APPS` section of the `base.py` file in the `mysite/settings` directory. Look at this file to see how the `start` command has listed them in there.

Reviewer: Nice!

from django.db import models

> from wagtail.models import Page
> 
> # add these new imports
> from wagtail.fields import RichTextField
> from wagtail.admin.panels import FieldPanel
> 
> 
> # replace the existing class definition of 'HomePage' with this:

Reviewer: The intention here is that you should be able to copy-paste everything. Those comments wouldn’t be desirable to preserve in this code (they don’t meant anything outside the context of the step-by-step tutorial), so I don’t think this helps.

```suggestion
from wagtail.models import Page
from wagtail.fields import RichTextField
from wagtail.admin.panels import FieldPanel

```

can use any of the [Django core fields](https://docs.djangoproject.com/en/stable/ref/models/fields). `content_panels` define the
capabilities and the layout of the editing interface. When you add fields to `content_panels`, it enables them to be edited on the Wagtail interface. [More on creating Page models](../topics/pages).

Run:

> ```console
> $ python manage.py makemigrations # creates the migrations file
> $ python manage.py migrate # executes the migrations and updates the database with your model changes 

Reviewer: Nice! Let’s rework this a bit further so this is easier to copy-paste:

```suggestion
```sh
# Creates the migrations file.
python manage.py makemigrations
# Executes the migrations and updates the database with your model changes.
python manage.py migrate
```

```

You must run the above commands each time you make changes to the model definition.

You can now edit the homepage within the Wagtail admin area (on the side bar go to _Pages_ and click the edit button beside _Homepage_) to see the new body field. Enter some text into the body field, and publish the page by selecting _Publish_ at the bottom of the page editor, rather than _Save Draft_.

> The page template now needs to be updated to reflect the changes made to the model. Wagtail uses normal Django templates to render each page

Reviewer: This looks the same as before except for spacing?

```suggestion
The page template now needs to be updated to reflect the changes made
to the model. Wagtail uses normal Django templates to render each page
```

## A basic blog

> ```
> python manage.py startapp blog
> ```
> 
> to create a new app in your Wagtail site.
> 
> Add the new `blog` app to `INSTALLED_APPS` list in `mysite/settings/base.py`.

Reviewer: I don’t see what this brings compared to the previous version.

```suggestion
We are now ready to create a blog. To do so, run
`python manage.py startapp blog` to create a new app in your Wagtail project.

Add the new `blog` app to `INSTALLED_APPS` in `mysite/settings/base.py`.
```

to create a new app in your Wagtail site.

Add the new `blog` app to `INSTALLED_APPS` list in `mysite/settings/base.py`.

### Blog Index and Posts

> Let's start with a simple index page for our blog. In `blog/models.py`:

Reviewer: Not clear why we’d remove this.

    ]
```

Since you have made changes to `model.py` run the migrations:

```console
$ python manage.py makemigrations
$ python manage.py migrate
```

Since the model is called `BlogIndexPage`, the default template name
(unless we override it) will be `blog/templates/blog/blog_index_page.html`. Create this file

> with the following content:

Reviewer: This isn’t as good as what we’re replacing. The template name is just` blog_index_page.html`, not the full path, and information on where Django looks for it is valuable.

        InlinePanel('gallery_images', label="Gallery images"),
    ]

> # add a new BlogPageGalleryImage class

Reviewer: This feels redundant with the label above the code block.

{% endblock %}
```

Make sure to upload some images when editing the blog page on your Wagtail admin if you want them to be displayed after editing your blog page template.

Here we use the `{% image %}` tag (which exists in the `wagtailimages_tags` library, imported at the top of the template) to insert an `<img>` element, with a `fill-320x240` parameter to indicate that the image should be resized and cropped to fill a 320x240 rectangle. You can read more about using images in templates in the [docs](../topics/images).

!["Second Post" page, with title, date, intro, body, and a gallery of three images](../_static/images/tutorial/tutorial_6.png)

> Since our gallery images are database objects in their own right, we can now query and re-use them independently of the blog post body. Let's define a `main_image` method, which returns the image from the first gallery item (or `None` if no gallery items exist) in `blog/models.py`:

Reviewer: Since this is just the one method specifying where to add it doesn’t seem worthwhile to me? It has to be where `BlogPage` is defined to start with.

```suggestion
Since our gallery images are database objects in their own right, we can now query and re-use them independently of the blog post body. Let's define a `main_image` method, which returns the image from the first gallery item (or `None` if no gallery items exist):
```

You'll probably want to create the new page/view as a child of Homepage,
parallel to your Blog index. Give it the slug "tags" on the Promote tab.

> Add `/tags` to the server url in your browser and Django will tell you what you probably already knew:

Reviewer: ```suggestion
Access `/tags` and Django will tell you what you probably already knew:
```

This one could certainly be clearer but I don’t think the proposed change is clear enough either.
